packages/diagrams/frontend/sirius-components-diagrams/src/renderer/layout/layout.tsx

```diff
@@ -94,6 +88,7 @@ export const prepareLayoutArea = (
         id: `${node.id}-label-${index}`,
         key: `${node.id}-label-${index}`,
         role: 'button', // role applied by react flow
+        style: node.data.insideLabel?.overflowStrategy === 'NONE' ? undefined : { maxWidth: node.width },
```

Comment:
Why did you add this line?

Reply:
The pre-layout should as much as possible run in the same context as Reactflow.
Here, you are simulating the label inside a node.
I think we should not make any assumption here.

The label should be wrapped on the pre-layout of node.
That means we should never retrieve label bounds from label pre-layout but from node pre-layout.